### PR TITLE
Remove complicated asSeenFrom logic

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would try to calculate if a given symbol is actually a parent using a special locally calculated asSeenFromMap (used for generics). Now, since semanticdb gives us this information for free, we can remove the complex logic and also make it much more efficient.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala [hunk 1]
```diff
@@ -1,253 +1,50 @@
...
-    } else {
-      isVisiblySame && signaturesEqual(
-        otherSymbol.signature,
-        methodSymbolInfo.signature
-      )(
-        context
-      )
-    }
+  ): Boolean = {
+    methodSymbolInfo.overriddenSymbols.contains(parentSymbol.symbol)
```
Comment (tanishiking) [inline metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala:47]: 😍

File: metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala [hunk 2]
```diff
@@ -1,253 +1,50 @@
...
+      if isOverriddenMethod(methodSymbolInfo, parentSymbol)
     } yield methodSymbol
     validMethods.headOption
   }
 
   private def isOverriddenMethod(
       methodSymbolInfo: SymbolInformation,
-      otherSymbol: SymbolInformation,
+      parentSymbol: SymbolInformation,
       findParent: Boolean = false
```
Comment (tanishiking) [inline metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala:45]: unused parameter

File: metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala [hunk 3]
```diff
@@ -1,253 +1,50 @@
...
-import scala.meta.internal.semanticdb.ValueSignature
-import scala.meta.internal.semanticdb.WithType
 
 object MethodImplementation {
 
-  import ImplementationProvider._
-
   def findParentSymbol(
       bottomSymbolInformation: SymbolInformation,
       bottomClassSig: ClassSignature,
```
Comment (tanishiking) [inline metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala:10]: Unused parameter

**Review by tanishiking — APPROVED** [review]
I left some comments on some unused parameters, but LGTM!

File: metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala (hunk 2 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala]: Fixed!

File: metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala (hunk 3 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/implementation/MethodImplementation.scala]: Also fixed, thanks!

**Review by dos65 — APPROVED** [review]
LGTM!

**Pull request merged**